### PR TITLE
Feature/matching between users

### PR DESCRIPTION
**Pull request by imbieras**
Added the ability to "authenticate" with UserId, match with users.

### CODE REVIEW

File: StudyBuddy/Controllers/ProfileController.cs [hunk 1]
```diff
@@ -112,8 +118,87 @@ public async Task<IActionResult> SaveProfile(string name, string birthdate, stri
...
+    {
+        UserId? currentUserId = _userService.GetCurrentUserId();
+
+        if (currentUserId != null && _userManager.GetUserById(currentUserId.Value) != null)
+        {
+            return RedirectToAction("Index", "Home");
+        }
+
+        if (string.IsNullOrEmpty(userId))
+        {
```
Comment (canyonhhh) [inline StudyBuddy/Controllers/ProfileController.cs:136]: `TempData["ErrorMessage"] = "User ID not provided";` maybe?

File: StudyBuddy/Managers/UserManager.cs [hunk 2]
```diff
@@ -9,6 +8,9 @@ namespace StudyBuddy.Managers;
 public class UserManager : IUserManager
 {
     private static readonly List<User?> s_users = new();
+    private readonly Dictionary<UserId, List<UserId>?> _matches = new();
```
Comment (canyonhhh) [inline StudyBuddy/Managers/UserManager.cs:11]: Duplicate

**Review by canyonhhh — CHANGES_REQUESTED** [review]

File: StudyBuddy/Managers/UserManager.cs (hunk 2 above)
Comment (imbieras) [reply StudyBuddy/Managers/UserManager.cs:11]: That doesn't even exist?

File: StudyBuddy/Controllers/ProfileController.cs (hunk 1 above)
Comment (imbieras) [reply StudyBuddy/Controllers/ProfileController.cs:136]: This would cause the error to pop up every time you enter the page as the textbox is empty.

**Review by canyonhhh — APPROVED** [review]

**Pull request merged**